gax-httpjson/src/main/java/com/google/api/gax/httpjson/EmptyMessage.java

```diff
@@ -58,4 +59,62 @@ public Object getFieldValue(String fieldName) {
   public ApiMessage getApiMessageRequestBody() {
     return null;
   }
+
+  @Override
+  public boolean equals(Object other) {
+    return other instanceof EmptyMessage;
+  }
+
+  private static final EmptyMessage DEFAULT_INSTANCE;
+
+  public static EmptyMessage getDefaultInstance() {
+    return DEFAULT_INSTANCE;
+  }
+
+  private EmptyMessage() {}
+
+  public static Builder newBuilder() {
+    return DEFAULT_INSTANCE.toBuilder();
+  }
+
+  public static Builder newBuilder(EmptyMessage prototype) {
+    return DEFAULT_INSTANCE.toBuilder().mergeFrom(prototype);
+  }
+
+  public Builder toBuilder() {
+    return this == DEFAULT_INSTANCE ? new Builder() : new Builder().mergeFrom(this);
+  }
+
+  static {
+    DEFAULT_INSTANCE = new EmptyMessage();
+  }
+
+  public static class Builder {
+    Builder() {}
+
+    public Builder mergeFrom(EmptyMessage other) {
+      return this;
+    }
+
+    public Builder(EmptyMessage other) {}
+
+    public EmptyMessage build() {
+      return new EmptyMessage();
```

Comment:
It seems you want to enforce singleton on `EmptyMessage`, then probably:
- make the constructor private
- change `new EmptyMessage()` to `DEFAULT_INSTANCE`, since we want to make sure there is only one instance of this class
- probably make a `DEFAULT_INSTANCE` for `Builder`. If there will be only one instance of `EmptyMessage`, then we probably only need one instance of `Builder`.

Reply:
I wanted this to emulate the com.google.protobuf.Empty java class.

Reply:
It's not that EmptyMessage should be a singleton, only the DEFAULT_INSTANCE.

Reply:
I think it's OK to make `EmptyMessage` a singleton, because `EmptyMessage` does not have any fields, and thus all the instances of this class would be the same.

Reply:
Done.